FiveOhFirstDataCore/Pages/Utility/UploadPFP.razor

```diff
@@ -0,0 +1,29 @@
+﻿@page "/utility/uploadpfp"
+@inject IUploadService _upload
+<h1>Upload a Profile Picture</h1>
+<DynamicAuthorizeView SectionName="UploadPFP">
+    <Authorized>
+        <div class="rounded bg-secondary pfp">
+            @if(pfp is not null && pfp.Length > 1)
+            {
+                <img class="rounded" src="data:image/png;base64, @(Convert.ToBase64String(pfp))"/>
+            } 
+            else if(CurrentUser is not null && CurrentUser.PFP is not null && CurrentUser.PFP.Length > 1)
+            {
+                <img class="rounded" src="data:image/png;base64, @(Convert.ToBase64String(@CurrentUser.PFP))"/>
+            }
+            else
+            {
+                <img class="rounded" src="https://www.501stlegion-a3.com/media/img/900x600/2021-04/501stlogo__1_.png"/>
+            }
+        </div>
+        <InputFile class="button" id="UploadPFP" OnChange="@Upload" accept=".png" /><br />
+        @if (BytesRead > 1)
+        {
+            <i>@((BytesRead/1024).ToString("0.##KB")) Uploaded Successfully</i>
+        }
+    </Authorized>
+    <NotAuthorized>
+        <h1>Welcome Guest!</h1>
```

Comment:
This should be set to a `<NotAuthorizedMessage />`

Reply:
Oh okay, I copied this from the index page, does that mean Authorized should be AuthorizedMessage?

Reply:
No, `<NotAuthorizedMessage />` just displays a generic not authorized message. It would still be encapsulated in the `<NotAuthorized>` block.